Sources/Purchasing/Configuration.swift

```diff
@@ -129,9 +129,23 @@ import Foundation
          * Set `observerMode`.
          * - Parameter observerMode: Set this to `true` if you have your own IAP implementation and want to use only
          * RevenueCat's backend. Default is `false`.
+         *
+         * - Note: This assumes your IAP implementation uses StoreKit 1.
+         * If you use StoreKit 2, use ``with(observerMode:storeKit2:)`` instead.
          */
         @objc public func with(observerMode: Bool) -> Builder {
+            return self.with(observerMode: observerMode, storeKit2: false)
+        }
+
+        /**
+         * Set `observerMode` with a corresponding `StoreKit 2` setting.
+         * - Parameter observerMode: Set this to `true` if you have your own IAP implementation and want to use only
+         * RevenueCat's backend. Default is `false`.
+         * - Parameter storeKit2: Set this to `true` if your own IAP implementation uses StoreKit 2. Default is `false`.
+         */
+        @objc public func with(observerMode: Bool, storeKit2: Bool) -> Builder {
             self.observerMode = observerMode
+            self.storeKit2Setting = .init(useStoreKit2IfAvailable: storeKit2)
```

Comment:
I'm wondering whether we really need to keep `useStoreKit2IfAvailable` deprecated... Considering we still need to use it here. Also, devs can potentially do `.with(observerMode: false, storeKit2: true)`, to bypass the deprecation and achieve the same result, so it's pretty much as if it wasn't deprecated...

Reply:
> Also, devs can potentially do .with(observerMode: false, storeKit2: true)

Yeah maybe the method should have never been `.with(observerMode: Bool)` but instead `.enableObserverMode()`.

Reply:
Feels a little odd to have to select "storekit 2 false" now, like, I'm not sure that tells you all that much about what's happening. 

How about 

```swift

enum StoreKitVersion {
    case StoreKit1, StoreKit2
}

.with(observerMode: Bool, storeKitVersion: StoreKitVersion) -> Builder {
```

(well, with changes for objc compatibility for the enum and stuff)

Reply:
Isn't that the same for `usesStoreKit2IfAvailable`? We made that a `Bool` too.
While we're at it, we could add an overload of that one to use this new enum. What do you think?

Reply:
> we could add an overload of that one to use this new enum
to the deprecated one? would the overload also be deprecated?

Reply:
Yeah, both. It would also have to need a new name, probably `.with(storeKit2Version:)`

Reply:
I wouldn't add a new API that's already deprecated when we add it. 
I was only suggesting that 

```swift
.with(observerMode: false, storeKitVersion: storeKit1)
```

might be slightly more expressive than
```swift
.with(observerMode: false, storeKit2: false)
```

Reply:
That way the docstring can read something like 

```swift
* Set `observerMode` with a corresponding `StoreKit` implementation
* - Parameter storeKitVersion: Set this to the StoreKit implementation 
* your app uses for purchases. 
* I.e.: if your app uses StoreKit 1, set it to `.storeKit1`, 
* and if your app uses StoreKit 2, set it to `.storeKit2`. 
* Apps using StoreKit 1 use `SKPaymentQueue` for transactions, 
* whereas StoreKit 2 uses `StoreKit.Product.Purchase`. 
```

Added some more text in case folks don't know which is which. Apple removed many of the SK1 docs, so I think it's likely that folks will start getting confused about this over time as SK1 slowly fades away

Reply:
I'd avoid adding a new enum + docs + Obj-C equivalent etc just for one API.
But I do think it makes sense to change the deprecated method (which arguably won't be once we fully support SK2) to use it as well.

Reply:
> I'd avoid adding a new enum + docs + Obj-C equivalent etc just for one API.
But I do think it makes sense to change the deprecated method (which arguably won't be once we fully support SK2) to use it as well.

I missed this comment but that makes a lot of sense